### PR TITLE
feat(openchallenges): implement input data type filter on challenge search page

### LINKED ISSUE #2578: [Feature] Create a filter component for challenge input data types
### What product(s) is this feature for?

OpenChallenges

### Description

- Enable the user to visualize all the input data types that exist
- Enable to configure the filter so that only the input data types that are associated to at least one challenge are visible
- Enable the user to type keywords in an input field to filter further input data types
- The filter component/service generates an array of data types ID (e.g. EDAM concept class id) that can later be used to filter challenges

To discuss: how could we represent the hierarchical relationship between different data types since EDAM support this? Identify the pros and cons (potentially more difficult to implement; less user-friendly?)

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by rrchai**
- closes #2578 

The filter should now be able to search the EDAM data concepts and be used to filter challenges.

## Changelogs
- [x] enable to list out EDAM data results (sorted alphabetically) in the input data type filter
- [x] enable to search data type by keywords in the search bar of the filter
- [x] enable to filter challenges by input data type keywords in the search bar of the page
- [x] enable to filter challenges by selected input data type(s)

## Preview
Recording 2024-04-24 at 11 29 21

---
There is a bug I notice for the search dropdown filter - tracked https://github.com/Sage-Bionetworks/sage-monorepo/issues/2648, and will be fixed in a future pr.

### CODE REVIEW

**Review by gaiaandreoletti — APPROVED** [review]

**Review by tschaffter — COMMENTED** [review]
The new component works as described.

How does the bug that you mention and are fixing in another PR affects the shipping of the present PR?

- Do you plan to merge the second PR into this present PR, then merge it to `main`?
- Do you want to merge this PR into `main` first, then merge the second PR to `main`?
  - This is OK if you consider that the first PR could potentially be shipped to prod without the fix. We could also wait for the fix to be ready before merging the two PRs consecutively.

One concept that may help here and for sure in the future are "feature flags". Here, we could show/hide the new UI component using a boolean flag in the app config. We can talk more about this strategy next time we meet.

**Comment by rrchai** [comment]
>How does the bug that you mention and are fixing in another PR affects the shipping of the present PR?
> - Do you plan to merge the second PR into this present PR, then merge it to main?
> - Do you want to merge this PR into main first, then merge the second PR to main?

I prefer to merge this PR first once approved, since this bug has already been there in prod on other filter like `organization`, and then I will fix the bug in a separated PR.

**Pull request merged**
